### PR TITLE
8005 heatmap style

### LINKED ISSUE #8005: [STYLE] Resolution for heatmaps is wrong
It is 16px by default.
For heatmaps I'd put 4 or even 2 by default.

Also the resolution parameter should be outside the animated option on this aggregation mode.

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #8005.

resolution-style

Also, it fixes the bug related with the scroll not updating the position to show the selected item.

cc @xavijam

### CODE REVIEW

**Comment by javisantana** [comment]
this should be reviewed by @makella

File: lib/assets/javascripts/cartodb3/editor/style/style-defaults/heatmap-style-defaults.js [hunk 1]
```diff
@@ -38,7 +45,7 @@ module.exports = _.defaults({
         overlap: false,
         duration: 30,
         steps: 256,
-        resolution: 2,
+        // resolution: 2,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/style/style-defaults/heatmap-style-defaults.js]: Remove it.

File: lib/assets/javascripts/cartodb3/editor/style/style-form/style-properties-form/style-shape-properties-form-model.js [hunk 2]
```diff
@@ -23,6 +24,8 @@ module.exports = StyleFormDefaultModel.extend({
 
     if (r.type === 'heatmap') {
       attrs = _.omit(attrs, 'stroke', 'blending');
+    } else {
+      attrs = _.omit(attrs, 'resolution');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/style/style-form/style-properties-form/style-shape-properties-form-model.js]: Test for this please :)

**Comment by xavijam** [comment]
Don't we need changes in the style generator part? I guess it should know where to find 'resolution' attribute cc @javisantana.

**Comment by saleiva** [comment]
I don't see the map updating on the animated gif, am I the only one?

**Comment by nobuti** [comment]
Fixed the generator part, also updated the gifcast.

File: lib/assets/javascripts/cartodb3/editor/style/style-converter.js [hunk 3]
```diff
@@ -308,14 +308,13 @@ var styleConversion = {
     if (animatedEnabled(style)) {
       style.animated = _.clone(_.defaults(style.animated || {}, defaultProperties));
     } else {
-      style.animated = _.extend({}, style.animated || {}, defaultProperties);
+      style.animated = _.extend({}, {resolution: style.resolution} || {}, defaultProperties);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/style/style-converter.js]: :+1:

**Comment by xavijam** [comment]
ready to go man, merge :es:

**Comment by makella** [comment]
@nobuti can you tell me a little more about the relationship between resolution and symbol size and how they work together?

we probably want to have a good default that works for few points and lots of points so any additional information you have would be super helpful.

thanks!

**Comment by nobuti** [comment]
@makella I'm not a big expert at all, but I'll try. As a Heatmap map is a variation of a Torque map, any of the Torque CartoCSS properties can be applied. Since Torque maps create a grid from your data and aggregates data to each cell of that grid, this property defines the width and height of each cell, in pixels. Larger numbers display larger grids of data. Here you have a great example about it: http://andrewxhill.com/blog/2015/04/17/torque-unknown/

**Comment by makella** [comment]
@nobuti 
totally. love that blog by andrew! Ok, thank you. I will spend some time with the new implementation and find a good default starting point. Thanks for the info!

**Pull request merged**